src/NHibernate/Impl/AbstractSessionImpl.cs

```diff
@@ -422,15 +466,15 @@ protected void EnlistInAmbientTransactionIfNeeded()
 
 		public void JoinTransaction()
 		{
-			CheckAndUpdateSessionStatus();
-			_factory.TransactionFactory.ExplicitJoinSystemTransaction(this);
+			using (BeginProcess())
+				_factory.TransactionFactory.ExplicitJoinSystemTransaction(this);
```

Comment:
Missing status check here. Having a closed session joining a transaction is an error.